### PR TITLE
Allow Structure.interpolate to extrapolate

### PR DESCRIPTION
**Pull request by kyledmiller**
## Summary

Major changes:

- added optional parameter to allow `Structure.interpolate()` to extrapolate beyond the `end_structure`. The parameter is the fractional amount of extrapolation beyond $q=1$ if $q$ is the normalized interpolation mode amplitude. Thus, `extrapolation=0.5` would produce structures spanning $q \in [0,1.5]$. This is helpful for capturing the outer piece of potential energy wells, e.g., when fitting a Landau model to an 2nd order structural phase transition.

## Todos

Create tests

## Checklist

- [x] Google format doc strings added. Check with `ruff`.
- [x] Type annotations included. Check with `mypy`.
- [ ] Tests added for new features/fixes.
- [x] If applicable, new classes/functions/modules have [`duecredit`](https://github.com/duecredit/duecredit) `@due.dcite` decorators to reference relevant papers by DOI ([example](https://github.com/materialsproject/pymatgen/blob/91dbe6ee9ed01d781a9388bf147648e20c6d58e0/pymatgen/core/lattice.py#L1168-L1172))

Tip: Install `pre-commit` hooks to auto-check types and linting before every commit:

```sh
pip install -U pre-commit
pre-commit install
```

### CODE REVIEW

File: pymatgen/core/structure.py [hunk 1]
```diff
@@ -2138,6 +2138,7 @@ def interpolate(
         interpolate_lattices: bool = False,
         pbc: bool = True,
         autosort_tol: float = 0,
+        extrapolation: float = 0,
```
Comment (janosh) [inline pymatgen/core/structure.py:2141]: Could you add a test for this new keyword?

**Review by janosh — CHANGES_REQUESTED** [review]

**Comment by kyledmiller** [comment]
Yep, It's on my to-do list. I'll try to get the tests written by the end of the week.

**Comment by kyledmiller** [comment]
Tests written

**Review by janosh — APPROVED** [review]
Tests look great! Just 1 line length issue left before we can merge.

**Comment by kyledmiller** [comment]
Looks good. I fixed that and also added some additions about cell setting & site order consistency to the docstring. If you don't think it belongs or should be in a separate PR, let me know.

**Pull request merged**